vendor/tooltipscan.lua

```diff
@@ -37,6 +37,50 @@ local function importTooltipTextToTable(tooltip, text, bag, slot)
     return tooltipTable
 end
 
+
+
+local function importTooltipColorToTable(tooltip, side, bag, slot)
+    -- We assume the tooltip is the GameTooltip
+    local tooltipText = "GameTooltipText"..side
+
+    -- If we don't have the gametooltip, use the scanning tooltip and use the bag item.
+    if not tooltip then
+        tooltip = scanningtip
+        tooltip:ClearLines()
+        tooltipText = "VendorScanningTipText"..side
+        if bag and slot then
+            tooltip:SetBagItem(bag, slot)
+        else
+            error("Invalid arguments to Tooltip Import")
+        end
+    end
+
+    -- Import the tooltip into a table
+    local tooltipTable = {}
+    for i=1, tooltip:NumLines() do
+        if _G[tooltipText..i] and _G[tooltipText..i]:GetText() then
+            local r, g, b = _G[tooltipText..i]:GetTextColor()
+
+            r = math.floor(r * 100 + 0.5) / 100
+            g = math.floor(g * 100 + 0.5) / 100
+            b = math.floor(b * 100 + 0.5) / 100
+
+            if r == 1 and g == 0.13 and b == 0.13 then
```

Comment:
Blizzard has well defined colors, e.g.:
RED_FONT_COLOR				= CreateColor(1.0, 0.1, 0.1);
The components are part of their font objects, e.g:
RED_FONT_COLOR.r, RED_FONT_COLOR.g, RED_FONT_COLOR.b.

All of them can be gotten with: RED_FONT_COLOR:GetRGBA()

From the look of several tooltips, they are using the RED_FONT_COLOR object when formatting the tooltip color: Examples:
GameTooltip:AddLine(BATTLE_PET_FIND_BATTLE_DISABLED, RED_FONT_COLOR.r, RED_FONT_COLOR.g, RED_FONT_COLOR.b, true);
self:AddTooltipLine(CLASS_TRIAL_RECOMMENDED_SPEC_ONLY, RED_FONT_COLOR);
GameTooltip:AddLine(TRANSMOGRIFY_INVALID_NO_ITEM, RED_FONT_COLOR.r, RED_FONT_COLOR.g, RED_FONT_COLOR.b, true);

This is sufficient evidence to me that proper detection of Red text would use the exact RED_FONT_COLOR object for comparison. And if for some reason blizzard changes their font color code, the addon will not break. We always prefer to use direct Blizzard variables where possible for this reason.

Reply:
They might use it to set the colors but the GetTextColor function returns whacky numbers (see picture) so you can't compare with that table.
https://i.imgur.com/WzvDbkl.png